src/components/select/select.js

```diff
@@ -0,0 +1,733 @@
+(function() {
+'use strict';
+
+/***************************************************
+
+### TODO ###
+
+**IMPLEMENTATION**
+
+- [x] Async loading of md-options (eg no options for one second, then they all show up).
+- [x] Style select button to match spec
+- [x] Implement CSS for an md-progress-circular in the select while the options are loading. 
+- [x] Implement CSS for md-optgroup
+- [X] Implement keyboard interaction
+- [X] Implement ARIA & accessibility
+- [x] Finish theming - theme color
+- [x] Add element option to interimElement to avoid recompiling every time (?)
+
+### TODO - POST RC1 ###
+- [ ] Abstract placement logic in $mdSelect service to $mdMenu service
+
+**DOCUMENTATION AND DEMOS**
+
+- [ ] ng-model with child mdOptions (basic)
+- [ ] ng-model="foo" ng-model-options="{ trackBy: '$value.id' }" for objects
+- [ ] mdOption with ng-value
+- [ ] mdOption with value
+- [ ] Multiple repeaters full of mdOptions
+- [ ] md-optgroups
+- [ ] Async fetching of options with a loader
+- [ ] Usage with input inside
+- [ ] Usage with custom <md-select-label> element inside
+- [ ] Usage with md-multiple
+
+***************************************************/
+
+var SELECT_EDGE_MARGIN = 8;
+var SELECT_PADDING = 8;
+var selectNextId = 0;
+
+angular.module('material.components.select', [
+  'material.core',
+  'material.components.backdrop'
+])
+.directive('mdSelect', SelectDirective)
+.directive('mdSelectMenu', SelectMenuDirective)
+.directive('mdOption', OptionDirective)
+.directive('mdOptgroup', OptgroupDirective)
+.provider('$mdSelect', SelectProvider);
+
+/**
+ * @ngdoc directive
+ * @name mdSelect
+ * @restrict E
+ * @module material.components.select
+ *
+ * @description Displays a select box, bound to an ng-model.
+ *
+ * @param {expression} ng-model The model!
+ * @param {boolean=} multiple Whether it's multiple.
+ */
+function SelectDirective($mdSelect, $mdUtil, $q, $mdTheming) {
+  return {
+    restrict: 'E',
+    compile: compile
+  };
+
+  function compile(element, attr) {
+    // The user is allowed to provide a label for the select as md-select-label child
+    var labelEl = element.find('md-select-label').remove();
+
+    // If not provided, we automatically make one
+    if (!labelEl.length) {
+      // Use the input as a label if there's an input inside.
+      if ( (labelEl = element.find('input')).length ) {
```

Comment:
@rschmukler any idea what this is supposed to be doing? Why would there be an input inside of `md-select`?